packages/models/src/errors.ts

```diff
@@ -98,12 +98,31 @@ const makeProblemLogString = (
   return `- title: ${problem.title} - detail: ${problem.detail} - errors: ${errorsString} - original error: ${originalError}`;
 };
 
+type ProblemBuilderOptions = {
+  /**
+   * If true, allows Problem objects received from downstream services
+   * to be passed through directly. If false, they are treated as internal errors.
+   *
+   * @default true
+   */
+  problemErrorsPassthrough?: boolean;
+  /**
+   * If true, replaces the details of errors that map to HTTP 500, or
+   * the ones that are passed through, with a generic message to avoid leaking internal information.
+   *
+   * @default false
+   */
+  hideInternalErrorDetails?: boolean;
```

Comment:
See comment above, I would not use the term internal, my proposal is `forceGenericProblemOn500` or something similar, so that the name explains what it does 
```suggestion
  forceGenericProblemOn500?: boolean;
```